src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,237 @@
+    }
+
+    if (callback) {
+        *callback = jetiExBusReadRawRC;
+    }
+
+    serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_RX_SERIAL);
+    if (!portConfig) {
+        return false;
+    }
+
+    serialPort_t *jetiExBusPort = openSerialPort(portConfig->identifier, FUNCTION_RX_SERIAL, jetiExBusDataReceive, baudRate, MODE_RX, SERIAL_NOT_INVERTED);
+
+    return jetiExBusPort != NULL;
+}
+
+
+void jetiExBusFrameReset(void)
+{
+	jetiExBusDataIncoming = false;
+    jetiExBusHeaderReady = false;
+	jetiExBusFramePosition = 0;
+	jetiExBusFrameLength = JETIEXBUS_MAX_FRAME_SIZE;
+}
+
+
+
+static uint16_t updateCRC16( uint16_t crc, uint8_t data ) 
+{
+	uint16_t ret_val;   
+	data ^= (uint8_t)(crc) & (uint8_t)(0xFF);   
+	data ^= data << 4;   
+	ret_val = ((((uint16_t)data << 8) | ((crc & 0xFF00) >> 8))             
+			  ^ (uint8_t)(data >> 4)
+			  ^ ((uint16_t)data << 3));   
+	return ret_val; 
+} 
+
+// Jeti Ex Bus CRC calculations
+uint16_t calcCRC16(volatile uint8_t *pt, uint8_t msgLen)
+{
+    uint16_t crc16_data=0;
+   	uint8_t mlen=0;
+    while(mlen<msgLen) {
+        crc16_data = updateCRC16(crc16_data, pt[mlen]);
+        mlen++;
+    }
+	return(crc16_data);
+}
+
+// Splits the received frame 
+void jetiExBusDecodeFrame()
+{
+    uint16_t value;
+    uint8_t frameAddr;
+
+	// Decode header
+	switch (((uint16_t)jetiExBusFrame[0] << 8)+((uint16_t)jetiExBusFrame[1])){
+		case (EXBUS_CHANNELDATA):
+		case (EXBUS_CHANNELDATA_TELEMETRY_REQUEST):
+			// Exctract the channeldata
+			for (uint8_t i = 0; i < jetiExBusChannelCount; i++){
+				frameAddr = JETIEXBUS_HEADER_LEN + i * 2;
+				value = ((uint16_t)jetiExBusFrame[frameAddr+1]) << 8;
+				value = value + ((uint16_t)jetiExBusFrame[frameAddr]);
+				// Convert to internal format
+				jetiExBusChannelData[i] = value>>3;
+				}
+			jetiExBusFrameReceived = true;
+			break;
+		case (EXBUS_DATA_REQUEST):
+			// Request for telemetry or JETIBOX menu
+			break;
+	}
+}
+
+
+
+// Receive ISR callback
+static void jetiExBusDataReceive(uint16_t c)
+{
+    uint32_t now;
+    static uint32_t jetiExBusTimeLast, jetiExBusTimeInterval;
+	uint16_t crc = 0;
+	
+    // Check if we shall reset frame position due to time
+    now = micros();
+    jetiExBusTimeInterval = now - jetiExBusTimeLast;
+    jetiExBusTimeLast = now;
+	
+    if (jetiExBusTimeInterval > JETIEXBUS_MAX_FRAME_TIME) {
+       jetiExBusFrameReset();
+    }
+    
+    // Check if we shall start a frame?
+    if (jetiExBusFramePosition == 0) {
+		if((c == JETIEXBUS_START_STICK_FRAME) || (c == JETIEXBUS_START_REQUEST_FRAME)){
+		    jetiExBusDataIncoming = true;
+		}
+    }
+
+    // Only do this if we are receiving to a frame
+    if (jetiExBusDataIncoming == true) {
+		// Store in frame copy
+        jetiExBusFrame[jetiExBusFramePosition] = (uint8_t)c;
+        jetiExBusFramePosition++;
+    }
+	
+	// Check the header for the messagelength
+	if((jetiExBusFramePosition==JETIEXBUS_HEADER_LEN) && (jetiExBusHeaderReady==false)) {
```

Comment:
The headerReady flag is redundant, `(jetiExBusFramePosition==JETIEXBUS_HEADER_LEN)` is true only once with current code